source/renderer/app/components/widgets/forms/PinCode.js

```diff
@@ -19,54 +21,281 @@ type Props = $Exact<{
   disabled: boolean,
   value: Array<string>,
   error: string | null,
+  selectedPinField: ?string,
 }>;
 
-export default class PinCode extends Component<Props> {
+type State = {
+  isBackSpace: boolean,
+  focusKeyChanged: boolean,
+  focusIsUpdated: boolean,
+};
+
+export default class PinCode extends Component<Props, State> {
   static defaultProps = {
-    length: 4,
+    length: VOTING_REGISTRATION_PIN_CODE_LENGTH,
     disabled: false,
     value: [],
   };
 
   inputsRef = [];
   focusKey = 0;
-  add = false;
+  isAddingNewValue = false;
+  fromBackspace = false;
+
+  state = {
+    isBackSpace: false,
+    focusKeyChanged: false,
+    focusIsUpdated: false,
+  };
+
+  valueHasChanged = (inputNewValue: string, key: number) => {
+    const { value } = this.props;
+    const emptyOrUnchangedValue = !value[key] || !inputNewValue;
+    const valueHasChanged =
+      inputNewValue &&
+      inputNewValue.length === 1 &&
+      value[key] !== inputNewValue;
+    return emptyOrUnchangedValue || valueHasChanged;
+  };
+
+  isInputValueSelected = (input: Field) => {
+    const { inputElement } = input;
+    const { current } = inputElement;
+    return current.selectionStart !== current.selectionEnd;
+  };
+
+  isInputFieldDisabled = (
+    index: number,
+    value: Array<string>,
+    disabled: boolean
+  ) => {
+    let inputFocusKey = 0;
+    const emptyFieldIndex = value.findIndex((item) => item === '');
+    if (emptyFieldIndex > -1 && !this.fromBackspace) {
+      inputFocusKey = emptyFieldIndex;
+    } else if (this.isAddingNewValue) {
+      inputFocusKey = this.focusKey + 1;
+    }
+    return (
+      disabled ||
+      (index > inputFocusKey && !value[index] && !value[inputFocusKey])
+    );
+  };
 
-  onChange = (inputValue: ?number, key: number) => {
+  onChange = (inputValue: ?number | ?string, key: number) => {
     const { value, onChange } = this.props;
+    const { isBackSpace } = this.state;
     const inputNewValue =
-      inputValue !== null && inputValue !== undefined
-        ? inputValue.toString()
-        : '';
-
-    if (
-      !Object.prototype.hasOwnProperty.call(value, key) ||
-      value[key] === '' ||
-      inputNewValue === ''
-    ) {
+      inputValue && !isNaN(inputValue) ? inputValue.toString() : '';
+    if (this.valueHasChanged(inputNewValue, key)) {
       const newValue = value;
-      newValue[key] = inputNewValue;
+      if (!isNaN(inputValue)) {
```

Comment:
`isNaN(new Number(STRING))` here as well

Reply:
This method shouldn't be accepting string values at all for inputValue, only numbers, hence i removed string type because it was obsolete.